R/get_fx_groups.R

```diff
@@ -78,111 +67,121 @@ get_fx_groups <- function(compound_sdf) {
+      nitrate = ChemmineR::smartsSearchOB(compound_sdf, nitrate_pattern),
+      nitro = ChemmineR::smartsSearchOB(compound_sdf, nitro_pattern),
+      amine_primary = groups$RNH2,
+      amine_secondary = groups$R2NH,
+      amine_tertiary = groups$R3N,
+      amine_aromatic = ChemmineR::smartsSearchOB(compound_sdf, amine_aromatic_pattern),
+      amide_primary = ChemmineR::smartsSearchOB(compound_sdf, amide_primary_pattern),
+      amide_secondary = ChemmineR::smartsSearchOB(compound_sdf, amide_secondary_pattern),
+      amide_tertiary = ChemmineR::smartsSearchOB(compound_sdf, amide_tertiary_pattern),
+      carbonylperoxynitrate = ChemmineR::smartsSearchOB(compound_sdf, carbonylperoxynitrate_pattern),
+      peroxide = ChemmineR::smartsSearchOB(compound_sdf, peroxide_pattern),
+      hydroperoxide = ChemmineR::smartsSearchOB(compound_sdf, hydroperoxide_pattern),
+      carbonylperoxyacid = ChemmineR::smartsSearchOB(compound_sdf, carbonylperoxyacid_pattern),
+      nitrophenol = ChemmineR::smartsSearchOB(compound_sdf, nitrophenol_pattern),
+      nitroester = ChemmineR::smartsSearchOB(compound_sdf, nitroester_pattern),
+      
+      # Additional groups from Meredith et al. 2023
+      phosphoric_acid = ChemmineR::smartsSearchOB(compound_sdf, phosphoric_acid_pattern),
+      phosphoric_ester = ChemmineR::smartsSearchOB(compound_sdf, phosphoric_ester_pattern),
+      sulfate = ChemmineR::smartsSearchOB(compound_sdf, sulfate_pattern),
+      sulfonate = ChemmineR::smartsSearchOB(compound_sdf, sulfonate_pattern),
+      thiol = ChemmineR::smartsSearchOB(compound_sdf, thiol_pattern),
+      carbothioester = ChemmineR::smartsSearchOB(compound_sdf, carbothioester_pattern),
+      oxygens   = atoms[["O"]] %||% 0L,
+      chlorines = atoms[["Cl"]] %||% 0L,
+      nitrogens = atoms[["N"]] %||% 0L,
+      sulfurs   = atoms[["S"]] %||% 0L,
+      phosphoruses = atoms[["P"]] %||% 0L,
+      bromines = atoms[["Br"]] %||% 0L,
+      iodines = atoms[["I"]] %||% 0L,
+      fluorines = atoms[["F"]] %||% 0L
+    ) %>% 
+    
+    # Corrections
+    dplyr::mutate(
+      #hydroperoxide pattern also picks up peroxyacids
+      hydroperoxide = .data$hydroperoxide - .data$carbonylperoxyacid,
+      #peroxide, nitrate, and ester patterns also pick up carbonylperoxynitrate group
+      ester = .data$ester - .data$carbonylperoxynitrate,
+      nitrate = .data$nitrate - .data$carbonylperoxynitrate,
+      peroxide = .data$peroxide - .data$carbonylperoxynitrate,
+      #phosphoric ester also matches phosphoric acid
+      phosphoric_acid = .data$phosphoric_acid - .data$phosphoric_ester,
+      
+      hydroxyl_aliphatic = .data$hydroxyl_total - .data$hydroxyl_aromatic, #TODO: should this happen before or after hydroyxl_aromatic is corrected? Or in the initial mutate?
+      
+      #according to SIMPOL.1 paper, nitrophenol shouldn't count aromatic hydroxyls.
+      hydroxyl_aromatic = .data$hydroxyl_aromatic - .data$nitrophenol,
```

Comment:
Currently tests are passing when things happen in this order, but I should double check some test cases and think about it.